src/rcheevos/operand.c

```diff
@@ -64,6 +65,54 @@ static int rc_parse_operand_lua(rc_operand_t* self, const char** memaddr, rc_par
   return RC_OK;
 }
 
+static int rc_parse_operand_variable(rc_operand_t* self, const char** memaddr/*, rc_parse_state_t* parse*/) {
+  const char* aux = *memaddr;
+  size_t i;
+  char varName[RC_VALUE_MAX_NAME_LENGTH + 1] = { 0 };
+
+  for (i = 0; i < RC_VALUE_MAX_NAME_LENGTH && *aux != '}'; i++) {
+    switch (*aux) {
+
+    default:
+      if (rc_is_valid_variable_character(*aux, i == 0)) {
+        varName[i] = *aux;
+        /* strncat_s(varName, RC_VALUE_MAX_NAME_LENGTH + 1, aux, 1); */
+        ++aux;
+      }
+      else {
+        ++aux;
```

Comment:
It's unnecessary to modify a local variable before an explicit return.

Also, recommend inverting the logic to make it more straight-forward:
```
if (!rc_is_valid_variable_character(*aux, i == 0))
  return RC_INVALID_VARIABLE_NAME;

varName[i] = *aux++;
break;
```

Reply:
I think I added the ++aux there it consume the } before returning. I'll review further once I have time to sit down and look.